examples/x86_64/src/hal.rs

```diff
@@ -0,0 +1,50 @@
+use core::{
+    ptr::NonNull,
+    sync::atomic::{AtomicUsize, Ordering},
+};
+use lazy_static::lazy_static;
+use log::trace;
+use virtio_drivers::{BufferDirection, Hal, PhysAddr, PAGE_SIZE};
+
+extern "C" {
+    fn end();
```

Comment:
Why is this a function rather than a static variable? I guess either works as you just want to get the address, but it seems a bit weird.